src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2152,9 +2153,10 @@ def _handle_upgrade_asm(self, new_profile: ServiceMeshProfile) -> Tuple[ServiceM
                     f"Please ensure all data plane workloads have been rolled over to revision {revision_to_keep} "
                     "so that they are still part of the mesh.\nAre you sure you want to proceed?"
                 )
-                if prompt_y_n(msg, default="y"):
-                    new_profile.istio.revisions.remove(revision_to_remove)
-                    updated = True
+                if not supress_confirmation and not prompt_y_n(msg, default="n"):
```

Comment:
I'm not sure I follow this line?

Reply:
`prompt_y_n(..)` is only run if `supress_confirmation` is false and we proceed based on `prompt_y_n(..)` return value.
if `supress_confirmation` is true, we skip this if statement.